prettier-loader.js

```diff
@@ -10,17 +11,27 @@ var promisedConfig = cosmiconfig('prettier')
 module.exports = function(source, map) {
   this.cacheable();
   var callback = this.async();
+  var ext = path.extname(this.resourcePath);
 
-  if (!new RegExp(this.query.test).test(this.context)) {
+  // .vue extension checking is a workaround for ignoring actual vue files when using vue-loader
+  if (!new RegExp(this.query.test).test(this.context) || ext === '.vue') {
     return callback(null, source, map);
   }
 
   promisedConfig.then(config => {
     var prettierSource;
     try {
+      var options = Object.assign({}, config, loaderUtils.getOptions(this));
+      if (ext === '.scss' || ext === '.sass') {
+        options.parser = 'scss';
+      } else if (ext === '.css') {
+        options.parser = 'css';
+      } else if (ext === '.less') {
+        options.parser = 'less';
+      }
```

Comment:
Yes, this is a great idea, but I think we should leave users an option to change parser to custom one, if they need it for listed extensions. So we must assign `parser` property inside `Object.assign(...)`  before `loaderUtils.getOptions(this)`.

Reply:
prettier current doesn't support any other parsers for these file types, so i figured this was not necessary for the time being. you would never have a `.less` or `.scss` file that you would want to parse with the `babylon` or `typescript` parser, for example--and this would in fact generate compiler errors anyways.

Reply:
Yes, you are right, but if someone wants to parse this files with some another parser different from postcss, that is builtin, I think it would be better to leave possibility to provide custom parser, because [prettier do](https://prettier.io/docs/en/options.html#parser).

And I think it would be more clear to user, if passed option will rewrite the default one.